### ISSUE TITLE
No Liskov violation reported in a situation where a non-generic method overrides a generic method

### ISSUE THREAD

**Issue by AlexWaygood** [microsoft/pyright#11128]
**Describe the bug**
Pyright does not report a Liskov violation on this code, whereas mypy and pyrefly both do:

```py
class A1:
    def method[T: int](self, x: T) -> T:
        return x

class B1(A1):
    def method(self, x: int) -> int:
        return int(x)
```

This means that pyright reports no errors on this larger snippet, and reveals the type on the final line as being `bool`, when in fact it is an `int` at runtime:

```py
class A1:
    def method[T: int](self, x: T) -> T:
        return x

class B1(A1):
    def method(self, x: int) -> int:
        return int(x)

def make_me_a_bool(a: A1) -> bool:
    return a.method(True)

reveal_type(make_me_a_bool(B1()))
```

**VS Code extension or command-line**

Playground: https://pyright-play.net/?pyrightVersion=1.1.405&pythonVersion=3.12&reportUnreachable=true&code=MYGwhgzhAECCCMAuAUNN0AmBTAZtAtlgC4AWA9hgNoAqi0AlgHZEC6AFBFiDgDTQAedagEpoAWgB80WqnRyATsQCu8xgOTJQkGACF4bBMJRzseQqQocuvAXSZFRkhs2Nz0ioirX22-YRtMCMABrLAB9QjCwMIAjMjIQNjA6Q3EpOITXNA8vaDAAOnNyDDZqeSUsf2RFADcsMBAwogBPAAcsNnwQ8MjojMS9NmFh5CA
